module/vision/GoalDetector/src/RansacGoalModel.h

```diff
@@ -24,69 +24,59 @@
 
 #include <array>
 #include "utility/math/geometry/Line.h"
+#include "utility/math/geometry/Plane.h"
 
 namespace module {
 namespace vision {
 
+    using Plane = utility::math::geometry::Plane<3>;
+
     class RansacGoalModel {
     public:
-        utility::math::geometry::Line left;
-        utility::math::geometry::Line right;
+        Plane leftPlane;
+        Plane rightPlane;
 
         static constexpr size_t REQUIRED_POINTS = 2;
 
         struct GoalSegment {
             GoalSegment() : left(arma::fill::zeros), right(arma::fill::zeros) {}
-            GoalSegment(const arma::vec2& l, const arma::vec2& r) : left(l), right(r) {}
-            arma::vec2 left;
-            arma::vec2 right;
+            GoalSegment(const arma::vec3& l, const arma::vec3& r) : left(l), right(r) {}
+            arma::vec3 left;
+            arma::vec3 right;
         };
 
         using DataPoint = GoalSegment;
 
-        RansacGoalModel() : left(), right() {}
+        RansacGoalModel() : leftPlane(), rightPlane() {}
 
         bool regenerate(const std::array<DataPoint, REQUIRED_POINTS>& pts);
 
         double calculateError(const DataPoint& p) const;
 
         template <typename Iterator>
         void refineModel(Iterator& begin, Iterator& end, const double& threshold) {
-
+            // TODO: new least squares model for plane
```

Comment:
This made a big difference in the old vision system to the quality of goal measurements.
Might be another PR though, make an issue for it if you don't do it now.